pkg/alertmanager/operator.go

```diff
@@ -920,6 +921,22 @@ receivers:
 	}
 
 	generator := newConfigGenerator(namespacedLogger, version, store)
+	// If defined global AlertmanagerConfig, find the global AlertmanagerConfig object from amConfigs.
+	// Otherwise work with the raw configuration secret.
```

Comment:
We need to get the object directly from the API since we already know its namespace and name (as I commented earlier, the object may not be in the map depending on the selectors).
And rather than deleting it from the map, `selectAlertmanagerConfigs()` can filter it out directly.

Reply:
fixed at [903c261](https://github.com/prometheus-operator/prometheus-operator/pull/4220/commits/903c2610d710f4d5c5ac574abfb0bae021a8db36)